### ISSUE TITLE
JavadocCommentsParser: quoted format specifier in {@value} tag is not parsed

### ISSUE THREAD

**Issue by Hajime-No-Ippo** [checkstyle/checkstyle#19001]
https://www.javaspecialists.eu/archive/Issue308-Formatting-Inlined-value-in-Javadocs.html described this javadoc new syntax

## Checkstyle version
13.3.0-SNAPSHOT

## Description
Since Java 20, the `{@value}` inline tag supports a quoted format string to control 
how a constant value is displayed. For example, from `java.lang.reflect.AccessFlag`:

```java
/**
 * The access flag {@code ACC_MANDATED} with a mask value of
 * <code>{@value "0x%04x" Modifier#MANDATED}</code>.
 */
public static final int MANDATED = 0x8000;
```

Running Checkstyle's Javadoc AST printer on this input throws a parse error:

```
java -jar checkstyle-13.3.0-SNAPSHOT-all.jar -j src/Test.java

Exception in thread "main" java.lang.IllegalArgumentException:
[ERROR:4] Javadoc comment at column 55 has parse error. Details:
mismatched input '%04x' expecting JAVADOC_INLINE_TAG_END while parsing INLINE_TAG
```

The current lexer rule in `JavadocCommentsLexer.g4` only supports bare unquoted format specifiers:

```
FORMAT_SPECIFIER
    : '%' [#+\- 0,(]* [0-9]* ('.' [0-9]+)? [a-zA-Z]
    ;
```

The grammar needs to be updated to support the quoted form (`"0x%04x"`) introduced in Java 20.

## Related
Part of #17882

**Comment by romani** [comment]
@Hajime-No-Ippo, please use markdown syntax to make issue looks good

**Comment by romani** [comment]
@Hajime-No-Ippo , please share whole compile able java file.
Please confirm that javadoc tool also processed such file without errors 

We need to know is this should be quoted always , or it is optional form.

**Comment by Hajime-No-Ippo** [comment]
@romani , 

Here's the compile able Test.java to this issue

**Quoted Form:**

```java
public class Test {
    /**
     * The access flag with a mask value of {@value "%04x" #MANDATED}.
     */
    public static final int MANDATED = 0x8000;
}
```

**Unquoted form:**

```java
public class Test {
    /**
     * The access flag with a mask value of {@value %04x #MANDATED}.
     */
    public static final int MANDATED = 0x8000;
}
```

Both versions pass javac and javadoc with 0 errors (only unrelated warnings about missing class comment). This means the quoted form is optional — both forms are accepted by the official javadoc tool.

```
javac src/Test.java && javadoc src/Test.java

Loading source file src/Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 25.0.1+8-LTS-27
Building tree for all the packages and classes...
Generating ./Test.html...
src/Test.java:1: warning: no comment
public class Test {
       ^
src/Test.java:1: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./constant-values.html...
Generating ./overview-tree.html...
Generating ./allclasses-index.html...
Building index for all classes...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
2 warnings
```

**Comment by romani** [comment]
>Unquoted form:

Still quoted:)

**Comment by Hajime-No-Ippo** [comment]
Sry, tested again, should fixed now

**Comment by AsthaDhapodkar** [comment]
Can I work on this?

**Comment by Hajime-No-Ippo** [comment]
@romani I'd like to work on this fix. Can I be assigned?

**Comment by romani** [comment]
No assignments. Contributors just make comment "I am on it" and start working 
I highly recommend to finish good xxxx issues first, to smoothly be onboarded into project, and you can do this issue after.

**Comment by AsthaDhapodkar** [comment]
I am on it.

**Comment by romani** [comment]
Will be fixed in a scope of https://github.com/checkstyle/checkstyle/issues/20391
